pkgs/servers/home-assistant/default.nix

```diff
@@ -176,6 +179,8 @@ in with py.pkgs; buildPythonApplication rec {
     "test_cached_event_message"
     # ValueError: count must be a positive integer (got 0)
     "test_media_view"
+    # AssertionError: len(events) == 1
+    "test_error_posted_as_event"
```

Comment:
```suggestion
    "test_error_posted_as_event"
    # No keyring backends are available
    "test_secrets_from_unrelated_fails"
    "test_secrets_credstash"
```

Reply:
Yeah, that sounds sensible. Will do!

There are so many tests that are failing and that I can't repro on my dev machines, it's a bit sad.

Reply:
So, the credstash one looks to be going away. I'm not so sure about `test_secrets_from_unrelated_fails`. Can you post the full traceback, please?

Reply:
```
homeassistant> ________________ TestSecrets.test_secrets_from_unrelated_fails _________________
homeassistant> [gw13] linux -- Python 3.8.6 /nix/store/papbnjfxwrdkmz6xfp7v2q7nxyvgcr5l-python3-3.8.6/bin/python3.8
homeassistant> self = <tests.util.yaml.test_init.TestSecrets testMethod=test_secrets_from_unrelated_fails>
homeassistant>     def test_secrets_from_unrelated_fails(self):
homeassistant>         """Test loading secrets from unrelated folder fails."""
homeassistant>         load_yaml(os.path.join(self._unrelated_path, yaml.SECRET_YAML), "test: failure")
homeassistant>         with pytest.raises(HomeAssistantError):
homeassistant> >           load_yaml(
homeassistant>                 os.path.join(self._sub_folder_path, "sub.yaml"),
homeassistant>                 "http:\n  api_password: !secret test",
homeassistant>             )
homeassistant> tests/util/yaml/test_init.py:393:
homeassistant> _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
homeassistant> tests/util/yaml/test_init.py:294: in load_yaml
homeassistant>     return load_yaml_config_file(fname)
homeassistant> homeassistant/config.py:329: in load_yaml_config_file
homeassistant>     conf_dict = load_yaml(config_path)
homeassistant> homeassistant/util/yaml/loader.py:62: in load_yaml
homeassistant>     return parse_yaml(conf_file)
homeassistant> homeassistant/util/yaml/loader.py:73: in parse_yaml
homeassistant>     return yaml.load(content, Loader=SafeLineLoader) or OrderedDict()
homeassistant> /nix/store/mvslgx44nxj2iyyy0cslj23da7c2zwax-python3.8-PyYAML-5.3.1/lib/python3.8/site-packages/yaml/__init__.py:114: in load
homeassistant>     return loader.get_single_data()
homeassistant> /nix/store/mvslgx44nxj2iyyy0cslj23da7c2zwax-python3.8-PyYAML-5.3.1/lib/python3.8/site-packages/yaml/constructor.py:51: in get_single_data
homeassistant>     return self.construct_document(node)
homeassistant> /nix/store/mvslgx44nxj2iyyy0cslj23da7c2zwax-python3.8-PyYAML-5.3.1/lib/python3.8/site-packages/yaml/constructor.py:55: in construct_document
homeassistant>     data = self.construct_object(node)
homeassistant> /nix/store/mvslgx44nxj2iyyy0cslj23da7c2zwax-python3.8-PyYAML-5.3.1/lib/python3.8/site-packages/yaml/constructor.py:100: in construct_object
homeassistant>     data = constructor(self, node)
homeassistant> homeassistant/util/yaml/loader.py:201: in _ordered_dict
homeassistant>     nodes = loader.construct_pairs(node)
homeassistant> /nix/store/mvslgx44nxj2iyyy0cslj23da7c2zwax-python3.8-PyYAML-5.3.1/lib/python3.8/site-packages/yaml/constructor.py:155: in construct_pairs
homeassistant>     value = self.construct_object(value_node, deep=deep)
homeassistant> /nix/store/mvslgx44nxj2iyyy0cslj23da7c2zwax-python3.8-PyYAML-5.3.1/lib/python3.8/site-packages/yaml/constructor.py:100: in construct_object
homeassistant>     data = constructor(self, node)
homeassistant> homeassistant/util/yaml/loader.py:201: in _ordered_dict
homeassistant>     nodes = loader.construct_pairs(node)
homeassistant> /nix/store/mvslgx44nxj2iyyy0cslj23da7c2zwax-python3.8-PyYAML-5.3.1/lib/python3.8/site-packages/yaml/constructor.py:155: in construct_pairs
homeassistant>     value = self.construct_object(value_node, deep=deep)
homeassistant> /nix/store/mvslgx44nxj2iyyy0cslj23da7c2zwax-python3.8-PyYAML-5.3.1/lib/python3.8/site-packages/yaml/constructor.py:100: in construct_object
homeassistant>     data = constructor(self, node)
homeassistant> homeassistant/util/yaml/loader.py:299: in secret_yaml
homeassistant>     pwd = keyring.get_password(_SECRET_NAMESPACE, node.value)
homeassistant> /nix/store/015wnml2z5y083ygx0dz4hfq9k8ls2c2-python3.8-keyring-21.5.0/lib/python3.8/site-packages/keyring/core.py:53: in get_password
homeassistant>     return _keyring_backend.get_password(service_name, username)
homeassistant> _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
homeassistant> self = <keyring.backends.fail.Keyring object at 0x7ffff44624c0>
homeassistant> service = 'homeassistant', username = 'test', password = None
homeassistant>     def get_password(self, service, username, password=None):
homeassistant>         msg = (
homeassistant>             "No recommended backend was available. Install a recommended 3rd "
homeassistant>             "party backend package; or, install the keyrings.alt package if "
homeassistant>             "you want to use the non-recommended backends. See "
homeassistant>             "https://pypi.org/project/keyring for details."
homeassistant>         )
homeassistant> >       raise NoKeyringError(msg)
homeassistant> E       keyring.errors.NoKeyringError: No recommended backend was available. Install a recommended 3rd party backend package; or, install the keyrings.alt package if you want to use the non-recommended backends. See https://pypi.org/project/keyring for details.
homeassistant> /nix/store/015wnml2z5y083ygx0dz4hfq9k8ls2c2-python3.8-keyring-21.5.0/lib/python3.8/site-packages/keyring/backends/fail.py:25: NoKeyringError
homeassistant> ----------------------------- Captured stderr call -----------------------------
homeassistant> DEBUG:tests.common:patch_yaml_files match /build/source/tests/testing_config/secrets.yaml
homeassistant> DEBUG:tests.common:patch_yaml_files match /build/source/tests/testing_config/configuration.yaml
homeassistant> DEBUG:homeassistant.util.yaml.loader:Loading /build/source/tests/testing_config/secrets.yaml
homeassistant> DEBUG:tests.common:patch_yaml_files match /build/source/tests/testing_config/secrets.yaml
homeassistant> DEBUG:homeassistant.util.yaml.loader:Secret http_pw retrieved from secrets.yaml in folder /build/source/tests/testing_config
homeassistant> DEBUG:homeassistant.util.yaml.loader:Secret comp1_un retrieved from secrets.yaml in folder /build/source/tests/testing_config
homeassistant> DEBUG:homeassistant.util.yaml.loader:Secret comp1_pw retrieved from secrets.yaml in folder /build/source/tests/testing_config
homeassistant> DEBUG:tests.common:patch_yaml_files match /build/source/tests/testing_config/unrelated/secrets.yaml
homeassistant> DEBUG:tests.common:patch_yaml_files match /build/source/tests/testing_config/subFolder/sub.yaml
homeassistant> DEBUG:homeassistant.util.yaml.loader:Loading /build/source/tests/testing_config/subFolder/secrets.yaml
homeassistant> DEBUG:homeassistant.util.yaml.loader:Loading /build/source/tests/secrets.yaml
homeassistant> DEBUG:homeassistant.util.yaml.loader:Loading /build/source/secrets.yaml
homeassistant> DEBUG:homeassistant.util.yaml.loader:Loading /build/secrets.yaml
homeassistant> ------------------------------ Captured log call -------------------------------
homeassistant> DEBUG    tests.common:common.py:714 patch_yaml_files match /build/source/tests/testing_config/secrets.yaml
homeassistant> DEBUG    tests.common:common.py:714 patch_yaml_files match /build/source/tests/testing_config/configuration.yaml
homeassistant> DEBUG    homeassistant.util.yaml.loader:loader.py:255 Loading /build/source/tests/testing_config/secrets.yaml
homeassistant> DEBUG    tests.common:common.py:714 patch_yaml_files match /build/source/tests/testing_config/secrets.yaml
homeassistant> DEBUG    homeassistant.util.yaml.loader:loader.py:283 Secret http_pw retrieved from secrets.yaml in folder /build/source/tests/testing_config
homeassistant> DEBUG    homeassistant.util.yaml.loader:loader.py:283 Secret comp1_un retrieved from secrets.yaml in folder /build/source/tests/testing_config
homeassistant> DEBUG    homeassistant.util.yaml.loader:loader.py:283 Secret comp1_pw retrieved from secrets.yaml in folder /build/source/tests/testing_config
homeassistant> DEBUG    tests.common:common.py:714 patch_yaml_files match /build/source/tests/testing_config/unrelated/secrets.yaml
homeassistant> DEBUG    tests.common:common.py:714 patch_yaml_files match /build/source/tests/testing_config/subFolder/sub.yaml
homeassistant> DEBUG    homeassistant.util.yaml.loader:loader.py:255 Loading /build/source/tests/testing_config/subFolder/secrets.yaml
homeassistant> DEBUG    homeassistant.util.yaml.loader:loader.py:255 Loading /build/source/tests/secrets.yaml
homeassistant> DEBUG    homeassistant.util.yaml.loader:loader.py:255 Loading /build/source/secrets.yaml
homeassistant> DEBUG    homeassistant.util.yaml.loader:loader.py:255 Loading /build/secrets.yaml
```

Reply:
On what target are you building this? I really wonder why I don't get any of these NoKeyringError exceptions. Are you on darwin?